core/config/project_settings.cpp

```diff
@@ -1053,7 +1053,9 @@ Error ProjectSettings::save_custom(const String &p_path, const CustomMap &p_cust
 		}
 	}
 	project_features = _trim_to_supported_features(project_features);
-	set_setting("application/config/features", project_features);
+	if (get_setting("application/config/features") != project_features) {
+		set_setting("application/config/features", project_features);
+	}
```

Comment:
```suggestion
	is_changed = true;
	set_setting("application/config/features", project_features);
	is_changed = false;
```
This is a hack, but it achieves the same (prevents another signal emitting) and is cheaper than fetching a property and comparing the arrays.